### DISCUSSION TITLE
How do I use enhanced tier of any model in a live transcription (websocket connection)?

### DISCUSSION THREAD

**Discussion by yashpalzala** [deepgram/community#97] [General help] [answered]
I'm trying to use enhanced tier of general model but I get error saying : 

`[ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: WebSocketChannelException: WebSocketChannelException: WebSocketException: Connection to 'https://api.deepgram.com:0/v1/listen?encoding=linear16&sample_rate=16000&language=en-IN&speech_final=true&model=general-enhanced#' was not upgraded to websocket`

My server url along with parameters is `"wss://api.deepgram.com/v1/listen?encoding=linear16&sample_rate=16000&language=en-IN&speech_final=true&model=general&tier=enhanced"`

The problem only occurs if I add the "tier" parameter.

**Comment by briancbarrow** [comment]
Hi @yashpalzala the `speech_final` parameter is something that will show up in the response object, not something that should be passed into the request. 

Do you still get errors if that is removed?

I just tested it using

**Reply by yashpalzala** [reply]
`speech_final` parameter is not the one causing the problem I removed it or keep it does not affect the outcome but when I put `tier=enhanced` as a parameter it causes the above given error. 

And i don't get error if I remove the tier parameter, but then how do I use enhanced tier of any model. 

I have followed the documentation here : 
```
By default, Deepgram applies its Base tier. If you would like to use a different tier, for a hosted deployment, add a tier parameter in the query string as well:

tier=enhanced&model=OPTION
```
Link to the documentation page: [https://developers.deepgram.com/documentation/features/model/](url)

**Comment by briancbarrow** [comment, marked as answer]
Hi @yashpalzala, Sorry I didn't catch this before now, but I just noticed you are using the `en-IN` language. Currently only the base tier is supported with that language code.
